department-of-veterans-affairs/caseflow
# [PART 1] Pass all tasks through queue config on the back end

## This PR
Bumps #14142

PART 1 of the stack
PART 2: #14491
PART 3: #14468
PART 4: #14486

### Description
Backend changes. Passes every task that should show in a user's queue through queue config rather than using QueueForRole. This will enable us to do 2 things:
- [ ] Stop using selectors on the front end. We'll be passing the exact tasks that queue tabs should show through queue config.
- [ ] Allow us to do partial pagination. As LegacyTasks only appear in judge and attorney "assigned" tabs, we can paginate their on hold and completed tabs without trying to paginate their assigned tab.

### Acceptance Criteria
- [ ] Tasks that should be shown in user queues are passed through queue config. (Note that as front end changes have not happened, the tasks that are currently shown in queues are pulled from QueueForRole and filtered and places in their correct tab via selectors. We want to ensure these match the ones that are selected for each tab on the back end and passed through queue config)

### Testing Plan
1. Disable pagination
   ```ruby
   FeatureToggle.disable!(:user_queue_pagination)
   ```
1. Log in as BVALSPORER
1. Go to your queue
1. Open up dev tools and ensure the tasks that are show in the queue tables match the tasks in queue config for each tab.
![Screen Shot 2020-06-04 at 12 56 20 PM](https://user-images.githubusercontent.com/45575454/83788387-1e630380-a663-11ea-9e27-2af5df993c2f.png)
![Screen Shot 2020-06-04 at 12 56 42 PM](https://user-images.githubusercontent.com/45575454/83788336-08edd980-a663-11ea-962f-e1627c677958.png)
![Screen Shot 2020-06-04 at 12 56 53 PM](https://user-images.githubusercontent.com/45575454/83788289-f7a4cd00-a662-11ea-8a87-ebee36211296.png)
1. Repeat the above for any attorney user. Filter out legacy appeal in their assigned tab. Ensure legacy colocated tasks are returned in the on hold tab.
![Screen Shot 2020-06-04 at 12 47 26 PM](https://user-images.githubusercontent.com/45575454/83787497-e3140500-a661-11ea-8049-9779f0615f99.png)
![Screen Shot 2020-06-04 at 12 48 09 PM](https://user-images.githubusercontent.com/45575454/83787851-6a617880-a662-11ea-94b3-7afab29d6653.png)
![Screen Shot 2020-06-04 at 12 48 27 PM](https://user-images.githubusercontent.com/45575454/83787917-8238fc80-a662-11ea-86a3-3447a30fac16.png)
1. Repeat the above for any judge user. Be sure to filter out legacy cases as those will not be pulled from queue config. Note that is only applies to their review page. Judge assign pages are not yet built from config.
![Screen Shot 2020-06-04 at 12 43 32 PM](https://user-images.githubusercontent.com/45575454/83787123-510bfc80-a661-11ea-823f-6efb6d704919.png)


### User Facing Changes
 - [ ] None (yet)

### Code Documentation Updates

## Full Stack
<details>
  <summary>SPOILERS!</summary>

### Tickets to write
- [x] Clean up front end task selectors, there are probably a few unused at this point, and we should remove them!
- [ ] Pass legacy tasks to the front end through queue config
- [ ] Build judge assign pages from config (would probably be a part of the "make judge teams act like orgs" planned epic)
</details>
